src/net/udp.rs

```diff
@@ -26,20 +28,19 @@ impl UdpSocket {
 
     pub fn recv(&mut self) -> io::Result<Option<Packet>> {
         let (len, _addr) = self.socket.recv_from(&mut self.recv_buffer)?;
-
         if len > 0 {
             // TODO: Remove unwrap and funnel result error types
             let raw_packet: RawPacket = deserialize(&self.recv_buffer[..len]).unwrap();
-            let packet = self.state.process_received(_addr, &raw_packet);
+            let packet = self.state.process_received(_addr, &raw_packet).unwrap();
             Ok(Some(packet))
         } else {
             Ok(None)
         }
     }
 
-    pub fn send(&mut self, packet: Packet) -> io::Result<usize> {
-        let (addr, payload) = self.state.pre_process_packet(packet);
-        self.socket.send_to(&payload, addr)
+    pub fn send(&mut self, packet: Packet) -> Result<io::Result<usize>, AmethystNetworkError> {
```

Comment:
The return type should be `Result<usize, AmethystNetworkError>`. Feel free to add the unwrap below, I want to implement the error stuff this weekend.

To get this currently working, youll need to implement:

```rust
impl From<std::io::Error> for AmethystNetworkError
```

Which is not done, but I plan on doing that this weekend and removing all unwraps.